pkg/e2e/operators/routemonitor.go

```diff
@@ -0,0 +1,126 @@
+
+	"github.com/openshift/osde2e/pkg/common/alert"
+	"github.com/openshift/osde2e/pkg/common/helper"
+
+	kubev1 "k8s.io/api/core/v1"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+var routeMonitorOperatorTestName string = "[Suite: informing] [OSD] Route Monitor Operator"
+
+func init() {
+	alert.RegisterGinkgoAlert(routeMonitorOperatorTestName, "SD-SREP", "Ron Green", "sd-cicd-alerts", "sd-cicd@redhat.com", 4)
+}
+
+var _ = ginkgo.Describe(routeMonitorOperatorTestName, func() {
+	const (
+		operatorNamespace = "openshift-route-monitor-operator"
+		operatorName      = "route-monitor-operator"
+		operatorLockFile  = "route-monitor-operator-lock"
+
+		defaultDesiredReplicas int32 = 1
+
+		consoleNamespace = "openshift-console"
+		consoleName      = "console"
+	)
+
+	var clusterRoles = []string{
+		"route-monitor-operator-admin",
+		"route-monitor-operator-edit",
+		"route-monitor-operator-view",
+	}
+
+	h := helper.New()
+
+	checkClusterServiceVersion(h, operatorNamespace, operatorName)
+	checkConfigMapLockfile(h, operatorNamespace, operatorLockFile)
+	checkDeployment(h, operatorNamespace, operatorName, defaultDesiredReplicas)
+	checkClusterRoles(h, clusterRoles, false)
+
+	// should I create a new helper here? seems everyone else does but I am not sure
+	checkUpgrade(helper.New(),
+		operatorNamespace,
+		operatorName,
+		operatorName,
+		"route-monitor-operator-registry")
+
+	ginkgo.Context("Route Monitor Operator regression", func() {
+		ginkgo.Context("console", func() {
+			ginkgo.It("has all of the required resouces", func() {
+				var err error
+				// RouteMonitor is expected to be there as a selectorsyncset
+				// see https://github.com/openshift/managed-cluster-config/blob/master/deploy/osd-route-monitor-operator/100-openshift-console.console.RouteMonitor.yaml
+				_, err = h.Prometheusop().MonitoringV1().ServiceMonitors(consoleNamespace).Get(context.TODO(), consoleName, metav1.GetOptions{})
+				Expect(err).NotTo(HaveOccurred(), "Could not get console serviceMonitor")
+				_, err = h.Prometheusop().MonitoringV1().PrometheusRules(consoleNamespace).Get(context.TODO(), consoleName, metav1.GetOptions{})
+				Expect(err).NotTo(HaveOccurred(), "Could not get console prometheusRule")
+			})
+		})
+	})
+	ginkgo.Context("Route Monitor Operator integration test", func() {
+		var (
+			routeMonitorNamespace = h.CurrentProject()
+			routeMonitorName      = "routemonitor-e2e-test"
+		)
+		ginkgo.It("has the required prerequisites for testing", func() {
+			var err error
+
+			//Launch a pod, service && route as the targets of the ServiceMonitor and PrometheusRules objects
+			pod := helper.SamplePod(routeMonitorName, routeMonitorNamespace, "quay.io/brancz/prometheus-example-app:v0.2.0")
+			err = helper.CreatePod(pod, routeMonitorNamespace, h)
+			Expect(err).NotTo(HaveOccurred(), "Couldn't create user workload monitoring testing pod")
```

Comment:
is this related to UWM?

Reply:
It's related to verifying metrics  and availability of a route..
I think this can also be used as a application which can receive route access for synthetic probes